src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,164 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
     @Override
     public void add(T value) {
+        linkLast(value);
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexPossibility(index);
+        if (index == size) {
+            linkLast(value);
+        } else {
+            linkBefore(value, getNodeByIndex(index));
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        if (list == null) {
+            throw new RuntimeException("The list is empty!");
+        }
+        for (var i : list) {
+            this.add(i);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndexExistence(index);
+        return getNodeByIndex(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndexExistence(index);
+        final Node<T> node = getNodeByIndex(index);
+        final T oldValue = node.value;
+        node.value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndexExistence(index);
+        return unlink(getNodeByIndex(index));
     }
 
     @Override
     public boolean remove(T object) {
+        for (Node<T> x = first; x != null; x = x.next) {
+            if ((x.value == object) || (x.value != null && x.value.equals(object))) {
+                unlink(x);
+                return true;
+            }
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void checkIndexPossibility(int index) {
```

Comment:
Use only one of `checkIndexPossibility` or `checkIndexExistence`, cause they contain completely the same logic. Also, rename the method simply as `checkIndex`.

Reply:
They contain different logic. Program will not work if only one of the methods is used.